### PR TITLE
(#252) Implement visual testing

### LINKED ISSUE #252: Build: Implement visual testing
From #211, we want to implement the following to automate visual regression testing:
1. Use Storybook to set up component templates
2. Use BackstopJS to automate tests
3. Update learna with new integration test directory (see spike/211-test-partials)
4. Create helpful developer documentation 

(Later on, we can add [Backbone to Cypress tests](https://github.com/garris/BackstopJS#integration-options-local-install).)

> **ESTIMATE** TBD

### Resources:
* Storybook: https://storybook.js.org/docs/react/get-started/introduction
* Backstop: https://github.com/garris/BackstopJS 
* Cypress implemntation notes:
  * https://github.com/NCIOCPL/clinical-trials-search-app/blob/develop/cypress/integration/TrialDescriptionPage/Accordion/index.js#L48
  * https://github.com/garris/BackstopJS#integration-options-local-install
* https://github.com/NCIOCPL/ncids/tree/spike/211-test-partials
* https://github.com/NCIOCPL/ncids/tree/ticket/211-shadow-storybook

### Notes
* See notes from #211

### PR DESCRIPTION
**Pull request by olitharp-nci**
Closes #252 

### Successful test:
https://github.com/NCIOCPL/ncids/actions/runs/1409322809

Artifact: [success-backstopjs.zip](https://github.com/NCIOCPL/ncids/files/7455664/success-backstopjs.zip)

### Failed test:
https://github.com/NCIOCPL/ncids/actions/runs/1409339745

Artifact: [failed-backstopjs.zip](https://github.com/NCIOCPL/ncids/files/7455666/failed-backstopjs.zip)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ NCIDS is built as a monorepo, utilizing *yarn* and *yarn workspaces* for depende
 | --------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | [`Documentation`](./docs)  | Design system documentation site                                                                                                                                                                                                       |
 | [`NCIDS-CSS`](./packages/ncids-css) | SASS/CSS package for NCIDS                                                                                                                                                                                                                              |
+| [`ncids-css-testing`](./packages/ncids-css-testing) | SASS/CSS testing package for NCIDS                                                                                                                                                                                                                              |
```
Comment (bryanpizzillo) [inline README.md:13]: Can you please move `ncids-css-testing` out of `packages` and into `testing`. If we have any automated stuff, we don't treat the testing package as a publishable thing,

File: packages/ncids-css-testing/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+## Update reference files
+1. To update existing files, you must run `yarn backstop:approve` with Storybook started
+
+## CI
+1. GitHub Actions will automatically run `yarn test:css:ci` for each push and pull request
+2. Failed jobs will be uploaded as an artifact for review
+   1. Download artifact
+   2. Open html-report/index.html
+3. Successful jobs will upload the test's `report.json` file
+
```
Comment (bryanpizzillo) [inline packages/ncids-css-testing/README.md:27]: Can you please add some notes as to what the scripts in engine-scripts are for, what they do, and where to find docs

File: packages/ncids-css-testing/__tests__/ncids-css-testing.test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+'use strict';
```
Comment (bryanpizzillo) [inline packages/ncids-css-testing/__tests__/ncids-css-testing.test.js:1]: Can you remove this please if it is not a test.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
This is looking really good. I have some comments below, mostly around updating documentation.

**Review by karlikpj — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**